module/VuFind/src/VuFind/RecordDriver/Feature/HierarchyAwareTrait.php

```diff
@@ -147,18 +147,18 @@ public function getContainingCollections()
                 break;
             case 'Top':
                 if (
-                    isset($this->fields['hierarchy_top_title'])
-                    && isset($this->fields['hierarchy_top_id'])
+                    !empty($this->getHierarchyTopTitle())
```

Comment:
You probably don't need the empty check -- just checking for truthiness should be sufficient.

I'd also suggest assigning these results to variables so you can reuse them in the logic below instead of calling the methods multiple times... e.g.:

```
$topTitles = $this->getHierarchyTopTitle();
$topIDs = $this->getHierarchyTopID();
if ($topTitles || $topIDs) {
```

Reply:
will do!